apps/prairielearn/src/lib/db-types.test.ts

```diff
@@ -0,0 +1,72 @@
+import { assert } from 'chai';
+
+import { IdSchema, IntervalSchema } from './db-types';
+
+describe('IdSchema', () => {
+  it('parses a valid id', () => {
+    const id = IdSchema.parse('123');
+    assert.equal(id, '123');
+  });
+
+  it('parses a nullable id', () => {
+    const id = IdSchema.nullable().parse(null);
+    assert.equal(id, null);
+  });
+
+  it('parses an optional id', () => {
+    const id = IdSchema.optional().parse(undefined);
+    assert.equal(id, undefined);
+  });
+
+  it('rejects a negative ID', () => {
+    const result = IdSchema.safeParse('-1');
+    assert.isFalse(result.success);
+  });
+
+  it('rejects a non-numeric ID', () => {
+    const result = IdSchema.safeParse('abc');
+    assert.isFalse(result.success);
+  });
+});
+
+describe('IntervalSchema', () => {
+  it('parses an interval with date', () => {
+    const interval = IntervalSchema.parse('1 year 2 months 3 days');
+    assert.equal(interval, 37000800000);
```

Comment:
I generated this and all the other expected values by running the intervals through `SELECT EXTRACT(epoch FROM ...)`.